src/averageAges.js

```diff
@@ -55,7 +73,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => person.sex === 'f')
+    .filter(mother =>
+      people.some(person => person.mother === mother.name));
+
+  const children = people
+    .filter(person => (onlyWithSon)
+      ? mothers.some(mother => person.mother === mother.name)
+        && person.sex === 'm'
+      : mothers.some(mother => person.mother === mother.name));
```

Comment:
and by the way, here you can filter by `people` array, and you won't need `mothers` anymore.